driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -726,8 +774,20 @@ public boolean onTimeout(Connection connection, long latency, int retryCount) {
             try {
                 connection.release();
 
-                logError(connection.address, timeoutException);
-                retry(false, null);
+                RetryPolicy retryPolicy = retryPolicy();
+                RetryPolicy.RetryDecision decision;
+                if (retryPolicy instanceof ExtendedRetryPolicy)
```

Comment:
Same as for the other comments about duplication, would be nice to have a function like `raiseClientTimenoutException()` to do it I think, but if you don't see the utility of it that's ok. It is just to make sure if we change the behaviour at some point about client timeout in the future, doing it in the function would make sure it is done for all places where we call `RetryPolicy.onClientTimeout()`